walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -63,52 +70,40 @@
+        solutions = modifySolutions(solutions, event, request.getDuration());
     }
     return solutions;
-    //cannot conflict with the collection of events given.
-    //make sure to check end point of time range is inclusive. 
+
   }
 
-  /**Splits a TimeRange into two time or shotens one TimeRange due to conflict with a given event */
+  /**Splits a TimeRange into two TimeRanges if conflicting event is contained in TimeRange
+   or shotens one TimeRange due to conflict with a given event */
   // Runtime: O(n) where n is the size of the Old Solutions.
-  private ArrayList<TimeRange> takeOutConflicts(ArrayList<TimeRange> oldSolutions, Event event, long duration) {
+  private ArrayList<TimeRange> modifySolutions(ArrayList<TimeRange> oldSolutions, 
+                                                                Event event, long duration) {
       ArrayList<TimeRange> solutions = new ArrayList<>();
       for (TimeRange slot: oldSolutions){
-          if (slot.contains(event.getWhen())){
-              //need to split slot into two
-              TimeRange slotA = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
-              TimeRange slotB = TimeRange.fromStartEnd(event.getWhen().end(), slot.end(), false);
-              //check if duration is still good for both slots.
-              if (checkTimeRangeisLongEnough(duration, slotA)){
-                  solutions.add(slotA);
-              }
-              if (checkTimeRangeisLongEnough(duration, slotB)){
-                  solutions.add(slotB);
-              }
+            // First make sure that conflicting event does not contian slot
+            // if it does, no longer want to add slot back into solutions.
+          if (event.getWhen().contains(slot)){
+              continue;
           }
-          else if (slot.overlaps(event.getWhen())){
-              // if event contains slot continue in the loop
-              if (event.getWhen().contains(slot)){
-                  continue;
-              }
-              //shorten slot in the front
-              else if (event.getWhen().end() > slot.start()){
-                   TimeRange shortenedSlot = TimeRange.fromStartEnd(event.getWhen().end(), slot.end(), false);
-                   if (checkTimeRangeisLongEnough(duration, shortenedSlot)){
-                       solutions.add(shortenedSlot);
-                   }
-              }
-              else{
-                  TimeRange shortenedSlot = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
-                  if (checkTimeRangeisLongEnough(duration, shortenedSlot)){
-                      solutions.add(shortenedSlot);
-                  }
-              }
-              
+          
+          else if (slot.overlaps(event.getWhen())){ 
+              // Need to take out slot and event overlap.
+            ArrayList<TimeRange> nonConflictingTimes = new ArrayList<>();
+            nonConflictingTimes.addAll(takeOutConflict(slot, event));
```

Comment:
You changed a similar circumstance to happen in one line, is there a reason this should be different?